legend-engine-application-query/src/main/java/org/finos/legend/engine/application/query/api/QueryStoreManager.java

```diff
@@ -200,17 +197,22 @@ public static void validateQuery(Query query)
         // TODO: we can potentially create a pattern check for version
     }
 
+    public List<Query> getQuerySearchResults(QuerySearchSpecification searchSpecification, List<Bson> filters)
+    {
+        return LazyIterate.collect(this.getQueryCollection()
+                .find(filters.isEmpty() ? EMPTY_FILTER : Filters.and(filters))
+                .sort(searchSpecification.showLatestQueriesFirst != null && searchSpecification.showLatestQueriesFirst ? descending("lastUpdatedAt") : EMPTY_FILTER)
+                // NOTE: return a light version of the query to save bandwidth
+                .projection(Projections.include("id", "name", "versionId", "groupId", "artifactId", "owner", "createdAt", "lastUpdatedAt"))
+                .limit(Math.min(MAX_NUMBER_OF_QUERIES, searchSpecification.limit == null ? Integer.MAX_VALUE : searchSpecification.limit)), QueryStoreManager::documentToQuery).toList();
+    }
+
     public List<Query> getQueries(QuerySearchSpecification searchSpecification, String currentUser)
     {
         List<Bson> filters = new ArrayList<>();
-        if (searchSpecification.showCurrentUserQueriesOnly != null && searchSpecification.showCurrentUserQueriesOnly)
-        {
-            // NOTE: every user is considered owner of the queries created by unknown user
-            filters.add(Filters.in("owner", currentUser, null));
-        }
         if (searchSpecification.searchTerm != null)
         {
-            filters.add(Filters.regex("name", Pattern.quote(searchSpecification.searchTerm), "i"));
+            filters.add(Filters.or(Filters.regex("name", Pattern.quote(searchSpecification.searchTerm), "i"), Filters.regex("id", Pattern.quote(searchSpecification.searchTerm), "i")));
```

Comment:
let's add doc here

Reply:
adding my comment in test -- I would like us to be a little more refined here, allowing search id like this would harbor mumbo jumbo in our search results, I think the best way is to check if the search term is of a particular format (i.e. (uuid)[https://en.wikipedia.org/wiki/Universally_unique_identifier] or something slightly looser), then we would proceed to do the ID matching